mypy/plugins/dataclasses.py

```diff
@@ -92,7 +92,7 @@ def transform(self) -> None:
             'frozen': _get_decorator_bool_argument(self._ctx, 'frozen', False),
         }
 
-        if decorator_arguments['init']:
+        if decorator_arguments['init'] and info.get('__init__') is None:
```

Comment:
This can get `__init__()` from `object`, this is why many tests failed. I think this should be simply `and '__init__' not in info.names`.

Reply:
@ilevkivskyi ugh not my brightest moment, fixed.